bitcoin/types_test.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2021 Rosetta Dogecoin Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bitcoin
+
+import (
+	"encoding/json"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_Block_UnmarshalJSON(t *testing.T) {
+	tests := map[string]struct {
+		json        string
+		expectedErr error
+	}{
+		"zero transactions": {
+			json:        `{ "tx": [] }`,
+			expectedErr: errors.New("expected >= 1 transactions in block, got 0"),
+		},
+		"one transaction": {
+			json: `{ "tx": [{}] }`,
```

Comment:
Thoughts on mocking out the actual `Transaction` JSON here instead of providing an empty object?

Reply:
@solo-fish I gave that a go with transaction JSON from the client fixtures. I could extract a separate fixture for this, but a standalone transaction doesn't seem to fit the mold of the existing fixtures, so I went with inlining it instead.

Reply:
I extracted a transaction type fixture for this, following the pattern used in `client_test.go`.

Reply:
Cool - thanks for raising a PR Travis :)